cmake/AwsFeatureTests.cmake

```diff
@@ -45,7 +45,8 @@ int main() {
     __asm__ __volatile__(\"\":\"=r\"(foo):\"r\"(bar):\"memory\");
 }" AWS_HAVE_GCC_INLINE_ASM)
 
-if(NOT LEGACY_COMPILER_SUPPORT)
+string(REGEX MATCH "^(aarch64|arm)" ARM_CPU ${CMAKE_SYSTEM_PROCESSOR})
+if(NOT LEGACY_COMPILER_SUPPORT OR ARM_CPU)
```

Comment:
I'm not super familiar with all our helper cmake files.
Is there another more appropriate place where we should declare that new ARM_CPU variable?

Reply:
I don't know either, it doesn't seem to be checked anywhere else.